### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- User can sign up, sign in, and sign out.
- User can only create an event and attend the event if signed in.
- User can see his past attended events.
- User can see all events without being authenticated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,73 @@
...
 
-* Database creation
+## You must have Ruby and Ruby on Rails installed on your system in order to run this  project locally.
 
-* Database initialization
+**Step 1**<br>
+Clone the repo in your local folder where you want to run it<br>
+`git@github.com:SajjadAhmad14/Private-events.git`.<br>
+**Step 2**<br>
+Run 'cd Private-events'<br>
```
Comment (lekegitrepo) [inline README.md:30]: - **_Add the `bundle install` command to the steps_**

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  include EventsHelper
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @user = current_user
+    @event = @user.created_events.new(event_params)
```
Comment (lekegitrepo) [inline app/controllers/events_controller.rb]: - **_Use `#build` method on association, i.e. `@user.created_events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
+
+  def index
+    @events = Event.all
+  end
```
Comment (lekegitrepo) [inline app/controllers/events_controller.rb]: - **_Should list past events, i.e. `@past_events = Event.past_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

- **_Should list ucomping events, i.e. `@upcoming_events = Event.future_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      redirect_to root_path
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb]: - **_Should list events created by user, i.e. `@created_events = @user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

- **_Should list events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

- **_Should list upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
#### Hi @SajjadAhmad14 :clap: :clap:
#### You have done a good job so far. However, you need to fix/implement the following requirements before this project can be approved. Check the review summary and the inline comments below :point_down:

#### App Behavior(When sign up as a new user no functionality is available) check it [here](https://www.loom.com/share/e3e8dd1d46a747cd9c8b7dfbff8b440f)
- **_Logged in user should be able to create new event **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_can_create_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

- _**Event's index page should display past and upcoming events lists ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))**_

- _**The event's show page should display the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))**_

- _**The user's show page should display events the user has created and events that the user is attending to separated in past and upcoming **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))**_

#### Controllers:
- **_Improve the create, show action methods of the event and user controller_**

#### Views and Helpers:
- **_Remove logic from the view use a helper method instead. For example, this code is not should not be in the view `current_user.attended_events.where("date < ?", Date.today)`_**

#### Tests:
- **_Add more tests and make sure all the tests pass(currently, some tests are failing)_**
* [ ] **_All the models should have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))_**
* [ ] **_Add integration tests for the event managment features (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))_**

#### _Reviewed by Temiayo Adeleke_

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Great job working on the requirements as stipulated by previous TSE.
However, I wasn't able to run the project due to this error
`SQLite3::SQLException: no such table: event_attences
/home/chubiyojo/.rvm/gems/ruby-2.6.3/gems/sqlite3-1.4.2/lib/sqlite3/database.rb:147:in `initialize'
`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi team! You've done a nice job working on the suggested changes from the earlier reviews. :clap: There are just a few issues for you to consider to complete this project:
- [ ] Your test suite has some failing tests. Please fix this.
- [ ] Add some tests for event management features e.g. creating an event. _[Hint]_ This could test authentication too since only logged-in users should be able to create an event. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] __[Optional]__ Make use of the `date_field` instead of the `text_field` for your date input on the form to create a new event. It is a more semantic option for dates.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED ⭐ 🥇 

Congratulations! You have made an amazing job and your project has been approved! 🥳 
Remember to fill up the completion form 📑  and you can move into the next project ⏩ 
Good luck! 😺

**Pull request merged**